src/compile.rs

```diff
@@ -329,6 +347,158 @@ pub fn build_func_map<'a>(
     Ok(decl_map)
 }
 
+pub fn validate_fs_context_duplicates(
+    fsc_rules: BTreeMap<&String, BTreeSet<&FileSystemContextRule>>,
+) -> Result<(), CascadeErrors> {
+    let mut errors = CascadeErrors::new();
+
+    'key_loop: for (_, v) in fsc_rules {
+        // We only have 1 or 0 elements, thus we cannot have a semi duplicate
+        if v.len() <= 1 {
+            continue;
+        }
+        let mut error: Option<CompileError> = None;
+        for rule in &v {
+            match rule.fscontext_type {
+                // If we ever see a duplicate of xattr task or trans we know something is wrong
+                FSContextType::XAttr | FSContextType::Task | FSContextType::Trans => {
+                    let mut range: Option<Range<usize>> = None;
+                    for arg in &rule.func_call.args {
+                        if let Argument::Quote(cas_arg) = &arg.0 {
+                            if *cas_arg == rule.fs_name {
+                                range = cas_arg.get_range();
+                            }
+                        }
+                    }
+                    error = Some(add_or_create_compile_error(error,
+                        "Duplicate filesystem context.",
+                        &rule.file,
+                        range,
+                        &format!("Found multiple different filesystem type declarations for filesystem: {}", rule.fs_name)));
+                }
+                FSContextType::GenFSCon => {
+                    // genfscon gets more complicated.  We can have similar rules as long as the paths are different.
+                    // If we find a genfscon with the same path, they must have the same context and object type.
+                    if let Some(path) = &rule.path {
+                        // Look through the rules again
+                        for inner_rule in &v {
+                            // Only check path if it was provided as part of the rule
+                            if let Some(inner_path) = &inner_rule.path {
+                                // If our paths match, check if our contexts match
+                                if path == inner_path && rule.context != inner_rule.context {
```

Comment:
The implication here is that duplicate genfscon rules are acceptable if the contexts are identical?

Reply:
Yes, my thought was that it is benign and doesn't directly add ambiguity to the final policy so let the user do it.